prov/shm/src/smr_ep.c

```diff
@@ -254,9 +255,24 @@ void smr_progress_resp(struct smr_ep *ep)
 		}
 		pending = container_of(dlist_entry, struct smr_pending_cmd, entry);
 
-		ret = ep->tx_comp(ep, (void *) resp->msg_id,
-				 (pending->cmd.hdr.op.op == ofi_op_tagged) ?
-				  FI_SEND | FI_TAGGED : FI_SEND, resp->status);
+		flags = 0;
+		switch (pending->cmd.hdr.op.op) {
+		case ofi_op_tagged:
+			flags = FI_TAGGED;
+			/* fall through */
+		case ofi_op_msg:
+			flags |= FI_SEND;
+			break;
+		case ofi_op_write:
+			flags = FI_RMA | FI_WRITE;
+			break;
+		case ofi_op_read_req:
+			flags = FI_RMA | FI_READ;
+			break;
+		default:
+			break;
+		}
```

Comment:
consider moving this switch statement to a new function to improve readability, e.g. smr_tx_comp_flags(op)